Geometry/TrackerNumberingBuilder/src/GeometricDet.cc

```diff
@@ -112,6 +114,12 @@ GeometricDet::GeometricDet(cms::DDFilteredView* fv, GeometricEnumType type)
       shape_(fv->shape()),
       params_(computeLegacyShapeParameters(shape_, fv->solid())),
       isFromDD4hep_(true) {
+  using namespace angle_units::operators;
+  if (almostEqual(phi_, -1._pi, 10)) {
```

Comment:
@cvuosalo - how much does it cost in terms of performance loss?

Reply:
@ianna , I agree with your concerns but just today we should decide to merge this PR or to close it. We cannot improve it anymore. 

I would propose to merge and  for the  next release 12_1_0 we should put efforts for the code optimization.

Reply:
These lines are executed when the `GeometricDet` is being built with DD4hep from XML. In production, the `GeometricDet` is read from the `PGeometricDet` in the DB, so this constructor is not called.
As a quick check, I looked at the times for the 11634.911 DD4hep workflow that reads geometry from XML. In the PR test for this PR (with the change) and the PR test for recent PR #34625, the average time per event in step1 simulation was 23 seconds in both cases, so there appears to be no large effect. Tests with larger numbers of events would be needed to see if there is a measurable effect.

Reply:
A quick check of the start-up time (including XML processing) of DD4hep XML workflow 11634.911 shows no significant difference between this PR and another PR test without this PR. More careful checks will be needed to determine whether this added function has a measurable effect on performance. Note that the `almostEqual` function is a simple template, so it shouldn't have much overhead.

Reply:
> These lines are executed when the GeometricDet is being built with DD4hep from XML. In production, the GeometricDet is read from the PGeometricDet in the DB, so this constructor is not called.

This is not true for the Phase 2 geometry that is using XML as a main source.

Reply:
> More careful checks will be needed to determine whether this added function has a measurable effect on performance. Note that the almostEqual function is a simple template, so it shouldn't have much overhead.

Yes, careful performance measurement ought to be done before merging the code. Any operation times the number of `GeometricDets` will have an effect.